src/Compiler/Checking/AttributeChecking.fs

```diff
@@ -461,7 +461,16 @@ let CheckFSharpAttributesForHidden g attribs =
 
 /// Indicate if a list of F# attributes contains 'ObsoleteAttribute'. Used to suppress the item in intellisense.
 let CheckFSharpAttributesForObsolete g attribs = 
-    not (isNil attribs) && (HasFSharpAttribute g g.attrib_SystemObsolete attribs)
+    not (isNil attribs) && 
+    (HasFSharpAttribute g g.attrib_SystemObsolete attribs) &&
+    // Exclude types marked with IsByRefLikeAttribute from being considered obsolete,
+    // even if ObsoleteAttribute is present. This avoids improper suppression of types 
+    // like Span and ReadOnlySpan in completion lists due to their dual attributes.
+    not (attribs |> List.exists (fun (Attrib(tcref, _, _, _, _, _, _)) -> 
```

Comment:
The test has failed.
I think this logic is super strange.
can you just follow the same level of asbtraction with the line above?

Since we have `HasFSharpAttribute g g.attrib_SystemObsolete attribs`, who not do `not(HasFSharpAttribute g g.attrib_ByRefLike attribs)` or similar?

Make sure TcGlobals has anything for IsByRefLike. If not, add it.
Make sure to treat that g.IsByRefLike as optional, since not every runtime has it !!!
i.e. it has to be an optional attribute reference in TcGlobals. Use it, if it is already there.